agent/pkg/proxy/proxy.go

```diff
@@ -59,8 +59,10 @@ func realServerAddress(conn *net.Conn) (string, int, error) {
 
 	file, err := tcpConn.File()
 	if err != nil {
+		tcpConn.Close()
```

Comment:
It doesn't need to close tcpConn here, because this function caller do that thing

Reply:
yes, you're right